### PR TITLE
Add Flatten and Reshape gradient tests

### PR DESCRIPTION
**Pull request by Shashi456**

### CODE REVIEW

**Comment by Shashi456** [comment]
Can someone trigger a build on this?

File: Tests/TensorFlowTests/LayerTests.swift [hunk 1]
```diff
@@ -639,6 +639,26 @@ final class LayerTests: XCTestCase {
         XCTAssertEqual(output.shape, expected)
     }
 
+    func testReshapeGradient() {
+        let layer = Reshape<Float>(shape: [10, 2, 1])
+        let input = Tensor(shape: [1, 5, 4], scalars: (0..<20).map(Float.init))
+        let computedGradient = gradient(at: input, layer) { $1($0).sum() }
+        // The expected value of the gradient was computed using the following Python code:
+        // ```
+        //   Reshape = tf.keras.layers.Reshape(target_shape = (10, 2, 1))
```
Comment (rxwei) [inline Tests/TensorFlowTests/LayerTests.swift]: Use camel case for Python variables.
```suggestion
        //   reshape = tf.keras.layers.Reshape(target_shape = (10, 2, 1))
```

File: Tests/TensorFlowTests/LayerTests.swift [hunk 2]
```diff
@@ -639,6 +639,26 @@ final class LayerTests: XCTestCase {
...
+        //   with tf.GradientTape() as t:
+        //     t.watch(x)
+        //     y = tf.math.reduce_sum(Reshape(x))
```
Comment (rxwei) [inline Tests/TensorFlowTests/LayerTests.swift]:
```suggestion
        //     y = tf.math.reduce_sum(reshape(x))
```

File: Tests/TensorFlowTests/LayerTests.swift [hunk 3]
```diff
@@ -647,6 +667,25 @@ final class LayerTests: XCTestCase {
         XCTAssertEqual(output.shape, expected)
     }
 
+    func testFlattenGradient() {
+        let layer = Flatten<Float>()
+        let input = Tensor(shape: [1, 4, 4], scalars: (0..<16).map(Float.init))
+        let computedGradient = gradient(at: input, layer) { $1($0).sum() }
+        // The expected value of the gradient was computed using the following Python code:
+        // ```
+        //   Flatten = tf.keras.layers.Flatten()
```
Comment (rxwei) [inline Tests/TensorFlowTests/LayerTests.swift]:
```suggestion
        //   flatten = tf.keras.layers.Flatten()
```

File: Tests/TensorFlowTests/LayerTests.swift [hunk 4]
```diff
@@ -647,6 +667,25 @@ final class LayerTests: XCTestCase {
...
+        //   with tf.GradientTape() as t:
+        //     t.watch(x)
+        //     y = tf.math.reduce_sum(Flatten(x)
```
Comment (rxwei) [inline Tests/TensorFlowTests/LayerTests.swift]:
```suggestion
        //     y = tf.math.reduce_sum(flatten(x))
```
Add a missing parenthesis.
